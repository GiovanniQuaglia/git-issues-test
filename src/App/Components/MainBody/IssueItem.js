import React from 'react';
import styled from 'styled-components';

const IssueItem = ({ issue }) => {
  const { title, html_url , number, user } = issue;
  return(
    <IssueWrapper>
      <Title href={html_url}>
        {title}
      </Title>
      <Details>
        {`#${number} created by ${user.login}`}
      </Details>
    </IssueWrapper>
  )
}

const IssueWrapper = styled.div`
  border-bottom: 1px solid lightgray;
  padding: 8px 0 0 16px;
  text-align: left;
  :hover {
    background-color: #F6F8FA;
    margin: 0 1px;
    padding: 8px 16px 0 16px;
  }
`;

const Title = styled.a`
  color: #24292e;
  font-weight: 600;
  text-decoration: none;
`;

const Details = styled.p`
  font-size: 12px;
  margin-top: 8px;
`;
export default IssueItem;