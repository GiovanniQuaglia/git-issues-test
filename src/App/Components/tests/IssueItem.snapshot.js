import React from 'react';
import renderer from 'react-test-renderer';
import IssueItem from '../MainBody/IssueItem';

const mockedProps = {  
  title: 'title',
  html_url: 'url',
  number: 0,
  user: 'user',
};

describe('IssueWrapper Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <IssueItem issue={mockedProps} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});