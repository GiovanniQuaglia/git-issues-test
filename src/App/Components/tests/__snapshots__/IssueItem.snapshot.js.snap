// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueWrapper Snapshot renders 1`] = `
<div
  className="sc-bdVaJa gvRItx"
>
  <a
    className="sc-bwzfXH kiOXrk"
    href="url"
  >
    title
  </a>
  <p
    className="sc-htpNat cQAwFE"
  >
    #0 created by undefined
  </p>
</div>
`;
